packages/gatsby-theme-store/src/sdk/minicart/Provider.tsx

```diff
@@ -0,0 +1,18 @@
+import React, { createContext, useCallback, useState } from 'react'
+import type { FC } from 'react'
+
+export interface IContext {
+  isOpen: boolean
+  toggle: () => void
+}
+
+export const Context = createContext<IContext | undefined>(undefined)
+
+export const Provider: FC = ({ children }) => {
+  const [isOpen, setIsOpen] = useState(false)
+  const toggle = useCallback(() => setIsOpen(!isOpen), [isOpen])
+
+  return (
+    <Context.Provider value={{ isOpen, toggle }}>{children}</Context.Provider>
```

Comment:
You should memo the value, otherwise, every time it renders will cause a re-render on children's that consumes the context, right?

Reply:
do we really? I thought react uses shallowCompare for state updates

Reply:
You're right! Sorry, i tried to replicate this behavior at the store and don't cause a re-render on a component that uses that context

Reply:
I get react shallowEqual algorithm and executed, you're right, shallowEqual return true for this object. I don't know why is re-rendering the context user